src/js/view/controls/components/menu/menu.js

```diff
@@ -270,7 +251,9 @@ export default class Menu extends Events {
                     return;
                 }
                 action(argument || index);
-                if (menuItem.deactivate) {  
+                // Re-assignment from index prevents side effect if action re-renders menu
+                const targetItem = this.items[index];
+                if (targetItem.deactivate) {  
```

Comment:
if this is the case, shouldn't we also replace `menuItem.active` above?

also i'm not clear on why a re-render would break these variables - it feels like the problem lies elsewhere

Reply:
I click the item -> It sets the level -> Before anything else executes, the 'levels' event is dispatched -> The menu is recreated and the items are replaced with the new items -> the old items function continues, and garbage isn't collected until the function is finished -> any item transformations are done on the orphaned nodes before they're garbage collected

Reply:
That's why it's assigned after action()

Reply:
Oh, just had an idea, one sec.